src/cargo/sources/registry/index.rs

```diff
@@ -204,6 +181,12 @@ impl<'cfg> RegistryIndex<'cfg> {
                 _ => true,
             }
         });
-        summaries.query(dep)
+
+        for summary in summaries {
```

Comment:
We can now remove `impl Registry for Vec<Summary>`, or implement it for an Iterator. I prefer dropping because this is an implementation detail, but providing an `impl` tricks the reader that `Vec<Summary>` is a real `Registry` which is used somewhere.